src/basic/sigbus.c

```diff
@@ -88,7 +90,7 @@ static void sigbus_handler(int sn, siginfo_t *si, void *data) {
 
         if (si->si_code != BUS_ADRERR || !si->si_addr) {
                 assert_se(sigaction(SIGBUS, &old_sigaction, NULL) == 0);
-                raise(SIGBUS);
+                rt_sigqueueinfo(getpid_cached(), SIGBUS, si);
```

Comment:
Those restrictions in the manpage are conditionalized on the following though:

> If the signal is being sent to any process other than the caller itself, the following restrictions apply ...

So that made me think it's fine here since we're sending the signal to ourselves.

Reply:
Ah gotcha that makes sense.

So I guess the other thing I'm not sure about is, `rt_sigqueueinfo()` could possibly hit the queue limit for real-time signals. Compared to `raise()` which sends a standard signal that doesn't queue. Does this matter?

Reply:
From reading the man pages, the way I understand it, Only real-time signals can be queued. Standard signals cannot be queued. Real-time signals are signals numbered from 32-63. The actual function used to send the signal doesn't affect whether it's a standard signal or a real-time signal. So since we're sending a standard signal using rt_sigqueueinfo(), it won't be queued at all just like before when we sent it with raise().